### ISSUE TITLE
The difference of stage 'pretrain', 'admm' and 'retrain'

### ISSUE THREAD

**Issue by chengwuxinlin** [yeshaokai/Robustness-Aware-Pruning-ADMM#5]
Inside the `run.sh.example`, it says

`#admm compression in adversarial setting
#python -u adv_main.py --config_file config.yaml --stage pretrain
#python -u adv_main.py --config_file config.yaml --stage admm
#python -u adv_main.py --config_file config.yaml --stage retrain`

From my understanding,  the `pretrain` stage will train a standard network, and `admm` stage will deploy weight pruning using ADMM and adversarial training on the standard network. So what is the stage `retrain` ? Do I need it to completely reproduce results in the paper?

**Comment by yeshaokai** [comment]
After admm stage, many weights are close to 0 sbut not really 0s. In the beginning of retrain, weights are forced to be 0s, causing performance degradation. Retrain stage masks thoese 0s and train the rest of weights. So retrain is necessary.
